packages/block-library/src/navigation-link/edit.js

```diff
@@ -302,6 +320,38 @@ export default function NavigationLinkEdit( {
 		transformToSubmenu,
 	] );
 
+	// Auto-clear URL and open Link UI when entity is not available and block is selected
+	// Keep binding so invalid state remains until user selects a new entity
+	useEffect( () => {
+		if (
+			isSelected &&
+			! isBoundEntityAvailable &&
+			id &&
+			metadata?.bindings?.url &&
+			url
+		) {
+			// Mark this change as not persistent to avoid creating undo levels
+			// for automatic cleanup of broken bindings
+			__unstableMarkNextChangeAsNotPersistent();
+			// Clear URL so user is forced to select a new entity
+			// Keep binding and ID so invalid state remains visible
+			setAttributes( {
+				url: undefined,
+			} );
```

Comment:
Why do we need to clear the invalid URL? I don't love seeing effects that set attributes. What's the harm in keeping it?

Reply:
The whole effect wasn't needed.